### PR TITLE
ghc 9.4 support

### PR DESCRIPTION
**Pull request by mikesperber**
Port to ghc 9.4.5.

### CODE REVIEW

**Comment by kenranunderscore** [comment]
I can confirm that the `concat` plugin work with GHC 9.4, based on these changes in our fork.
We are using a newer version though, 9.4.6 (and will probably be switching to 9.4.8 once/if it is released).

Don't know about the garnix and CI stuff, though. Maybe @sellout can chime in here? :)

(Also, am I imagining things, or didn't we have the 9.0 build failing before using `cabal v2-build`, but work with Nix, and now the CI seems to show the same (number of) failing tests, but for 9.2 :thinking: )

**Review by kenranunderscore — APPROVED** [review]

**Review by sellout — CHANGES_REQUESTED** [review]
It looks like the previous builds were killed for an unknown reason. I’ve restarted a couple to see how they go, but there’s at least one legit failure so far.

The GH workflow for 9.2 is also failing, but that’s due to gold tests, which I think were failing before this PR anyway?

File: plugin/src/ConCat/Plugin.hs [hunk 1]
```diff
@@ -1398,6 +1410,19 @@ mkOps (CccEnv {..}) guts annotations famEnvs dflags inScope evTy ev cat = Ops {.
 #else
    optimizeCoercion = optCoercion dflags emptyTCvSubst
 #endif
+    -- | Substitute new subexpressions for variables in an expression. Drop any dead
```
[the remark this answers is not in the document]
Comment (mikesperber) [reply plugin/src/ConCat/Plugin.hs]: I've changed it, but we'll have to see: I don't see this message locally.

**Comment by mikesperber** [comment]
> The GH workflow for 9.2 is also failing, but that’s due to gold tests, which I think were failing before this PR anyway?

Yes. There's some weird problem where building via Nix changes ghc's inlining strategy slightly as compared to Cabal. We'll have another look at it once this is merged.

File: plugin/src/ConCat/Plugin.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikesperber) [reply plugin/src/ConCat/Plugin.hs]: @sellout Haddock seems to have gone through now. OK to merge?

[the remark this answers is not in the document]
Comment (sellout) [reply plugin/src/ConCat/Plugin.hs]: Yep, this is now in a strictly better state than master.

master: 2 cancelled, 19 failing, and 49 successful checks
this PR: 73 successful, 1 cancelled, and 17 failing checks

(don’t know why GH is inconsistent with the ordering of successful/cancelled/failing)

I also removed the “required” flag from the GHC 9.2.4 checks that no longer exist, but didn’t add ones for 9.2.8. I figure until we have consistent passage, it’s more frustrating than helpful; and GH’s interface for changing required checks is painful (so might be worth adopting [Probot’s Settings app](https://probot.github.io/apps/settings/), which I’ve already done in some personal projects).

**Review by sellout — APPROVED** [review]

**Comment by kenranunderscore** [comment]
Just FYI, I'm quite certain that the "5 out of 83 tests" in the gold test suite failed on a different GHC version before (9.0) and has since "moved up" to now fail on 9.2 instead. Very weird; see https://github.com/compiling-to-categories/concat/pull/111#discussion_r1122078378

**Comment by mikesperber** [comment]
Moving this to master now.

**Pull request merged**